src/celengine/universe.cpp

```diff
@@ -717,16 +716,13 @@ Universe::getSolarSystem(const Selection& sel) const
 SolarSystem*
 Universe::createSolarSystem(Star* star) const
 {
-    SolarSystem* solarSystem = getSolarSystem(star);
-    if (solarSystem != nullptr)
-        return solarSystem;
-
-    solarSystem = new SolarSystem(star);
-    solarSystemCatalog->insert(SolarSystemCatalog::
-                               value_type(star->getIndex(),
-                                          solarSystem));
+    auto starNum = star->getIndex();
+    auto iter = solarSystemCatalog->lower_bound(starNum);
+    if (iter != solarSystemCatalog->end() && iter->first == starNum)
+        return iter->second.get();
 
-    return solarSystem;
+    iter = solarSystemCatalog->emplace_hint(iter, starNum, std::make_unique<SolarSystem>(star));
+    return iter->second.get();
```

Comment:
please comment this change in pr description

Reply:
done